contentmgr/replication.go

```diff
@@ -160,99 +160,36 @@ func (cm *ContentManager) newContentStagingZone(user uint, loc string) (*Content
 		return nil, err
 	}
 
-	return &ContentStagingZone{
-		ZoneOpened: content.CreatedAt,
-		MinSize:    cm.cfg.Content.MinSize,
-		MaxSize:    cm.cfg.Content.MaxSize,
-		User:       user,
-		ContID:     content.ID,
-		Location:   content.Location,
-		Readiness:  stagingZoneReadiness{false, "Readiness not yet evaluated"},
-	}, nil
-}
-
-func (cb *ContentStagingZone) updateReadiness() {
-	// if it's above the size requirement, go right ahead
-	if cb.CurSize > cb.MinSize {
-		cb.Readiness.IsReady = true
-		cb.Readiness.ReadinessReason = fmt.Sprintf(
-			"Current deal size of %d bytes is above minimum size requirement of %d bytes",
-			cb.CurSize,
-			cb.MinSize)
-		return
-	}
-
-	cb.Readiness.IsReady = false
-	cb.Readiness.ReadinessReason = fmt.Sprintf(
-		"Minimum size requirement not met (current: %d bytes, minimum: %d bytes)\n",
-		cb.CurSize,
-		cb.MinSize)
+	return content, nil
 }
 
-func (cm *ContentManager) tryAddContent(cb *ContentStagingZone, c util.Content) (bool, error) {
-	cb.lk.Lock()
-	defer cb.lk.Unlock()
-	defer cb.updateReadiness()
-
-	// TODO: propagate reason for failing to add content to bucket to end user
-
+func (cm *ContentManager) tryAddContent(zone util.Content, c util.Content) (bool, error) {
 	// if this bucket is being consolidated, do not add anymore content
-	if cb.IsConsolidating {
-		return false, nil
-	}
-
-	if cb.CurSize+c.Size > cb.MaxSize {
+	if cm.ZonesConsolidating[zone.ID] {
```

Comment:
dont you need to lock before check?

Reply:
yeah, you need the lock for each content add

Reply:
tryAddContent is only called while the bucketLk is already locked (https://github.com/application-research/estuary/pull/609/files#diff-20923025e6ed8064607b87fb913169095c8e675a484359bd37581febefc8295aL841)

can comment it's assuming the lock is already locked

but i added a couple other references to ZonesConsolidating that i need to wrap with the lock, will do now

Reply:
> tryAddContent is only called while the bucketLk is already locked

Ah, I see it now, nice - ignore my comment